src/recipe/recipe.controller.ts

```diff
@@ -1,8 +1,14 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
+import { RecipeDto } from './dto/recipe.dto';
 
 import { RecipeService } from './recipe.service';
 
 @Controller('recipe')
 export class RecipeController {
   constructor(private recipeService: RecipeService) {}
+
+  @Get('/search/:query')
+  searchRecipeTitles(@Param('query') query: string): Promise<RecipeDto[]> {
+    return this.recipeService.searchInTitle(query);
```

Comment:
`RecipeService#searchInTitle` allows you to pass pagination options, but you haven't exposed them in the endpoint. Not sure if this is by accident or if you don't want to expose them here and you always return the first page.